agent/connect/ca/provider_vault.go

```diff
@@ -976,57 +975,60 @@ func vaultLogin(client *vaultapi.Client, authMethod *structs.VaultAuthMethod) (*
 	return resp, nil
 }
 
-func configureVaultAuthMethod(authMethod *structs.VaultAuthMethod) (loginPath string, err error) {
+func configureVaultAuthMethod(authMethod *structs.VaultAuthMethod) (VaultAuthenticator, error) {
 	if authMethod.MountPath == "" {
 		authMethod.MountPath = authMethod.Type
 	}
 
+	loginPath := ""
 	switch authMethod.Type {
+	case VaultAuthMethodTypeAWS:
+		return NewAWSAuthClient(authMethod), nil
+	case VaultAuthMethodTypeGCP:
+		return NewGCPAuthClient(authMethod)
 	case VaultAuthMethodTypeKubernetes:
 		// For the Kubernetes Auth method, we will try to read the JWT token
 		// from the default service account file location if jwt was not provided.
 		if jwt, ok := authMethod.Params["jwt"]; !ok || jwt == "" {
 			serviceAccountToken, err := os.ReadFile(defaultK8SServiceAccountTokenPath)
 			if err != nil {
-				return "", err
+				return nil, err
 			}
 
 			authMethod.Params["jwt"] = string(serviceAccountToken)
 		}
-		loginPath = fmt.Sprintf("auth/%s/login", authMethod.MountPath)
+		return NewVaultAPIAuthClient(authMethod, loginPath), nil
```

Comment:
non-blocking: You don't have to do this here, but for organizational purposes, I wouldn't mind if we added functions for the other auth-method types that have non-default config: `NewK8sAuthClient`, `NewOCIAuthClient`, `NewAuthClientWithUsername`(?) - where those could check/set params and call `NewVaultAPIAuthClient`.

Reply:
Yes I fully agree. I've added a new issue to our backlog for this.